src/anaconda/.devcontainer/Dockerfile

```
@@ -3,9 +3,43 @@ FROM continuumio/anaconda3:2023.03-1 as upstream
 # Verify OS version is expected one
 RUN . /etc/os-release && if [ "${VERSION_CODENAME}" != "bullseye" ]; then exit 1; fi
 
+# Temporary: Upgrade python packages due to mentioned CVEs
+# They are installed by the base image (continuumio/anaconda3) which does not have the patch.
+RUN python3 -m pip install \
+    # https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-21797
+    --upgrade joblib \
+    # https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-24065
+    cookiecutter \
+    # https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-34749
+    mistune \
+    # https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-34141
+    numpy \
+    # https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-25577
+    werkzeug \
+    # https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-32862
+    nbconvert \
+   # https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-28370
+    tornado
+
+RUN conda install \ 
+    # https://github.com/advisories/GHSA-5cpq-8wj7-hf2v
+    pyopenssl=23.2.0 \ 
+    cryptography=41.0.2 \
+    # https://github.com/advisories/GHSA-j8r2-6x86-q33q
+    requests=2.31.0
+
 # Reset and copy updated files with updated privs to keep image size down
-FROM mcr.microsoft.com/devcontainers/base:0-bullseye
-COPY --from=upstream /opt /opt/
+FROM mcr.microsoft.com/devcontainers/base:1-bullseye
+
+ARG USERNAME=vscode
+
+# Create the conda group and add remote user to the group
+RUN groupadd -r conda --gid 900 \ 
+    && usermod -aG conda ${USERNAME}
+
+# Copy opt folder, set ownership and group permissions
+COPY --chown=:conda --chmod=775 --from=upstream /opt /opt/
```

```suggestion
COPY --chown=:conda --chmod=775 --from=upstream /opt/conda /opt/conda
```

> && chown -R :conda /opt/conda \
    && chmod -R g+w /opt/conda \
    
Syncing with ^

---

I agree that the current upstream image only adds `conda` folder to `/opt/`, however, that could change anytime. Hence, it would be nice to be as precise as possible!

---

Good point! I applied the suggested changes.

---

Related changes caused the permission denied error. I'm investigating why this happened.
Error: https://github.com/devcontainers/images/actions/runs/5760266653/job/15615884009#step:3:5724

---

The issue is related to the fact that when using chown and chmod functions with the `COPY` command, Docker updates the metadata of folder content but not the folder itself. I updated Dockerfile to set permissions to the `opt/conda` folder: https://github.com/devcontainers/images/pull/692/commits/ffece92eedb493a66f2cd3798efecbfa29c7a734.

Results:

> COPY /opt/conda /opt/conda # buildkit

```console
Permission     UID:GID       Size  Filetree
drwxr-xr-x         0:0     4.2 GB  └── opt 
drwxr-xr-x       0:900     4.2 GB      └── conda
```

> COPY /opt/ /opt/ # buildkit

```console
Permission     UID:GID       Size  Filetree
drwxr-xr-x         0:0     4.2 GB  └── opt
drwxrwxr-x       0:900     4.2 GB      └── conda
```

> COPY /opt/conda /opt/conda # buildkit
> RUN |1 USERNAME=vscode /bin/sh -c chmod =2775 /opt/conda # buildkit

```console
Permission     UID:GID       Size  Filetree
drwxr-xr-x         0:0        0 B  └── opt
drwxrwsr-x       0:900        0 B      └── conda
```

---

Nice, thanks!